raster_api/runtime/src/alternate_reader.py

```diff
@@ -0,0 +1,96 @@
+"""Custom Alternate Asset Reader."""
+
+from typing import Any, Dict, Optional, Set, Type
+
+import attr
+import pystac
+import rasterio
+from morecantile import TileMatrixSet
+from rasterio.crs import CRS
+from rio_tiler.constants import WEB_MERCATOR_TMS, WGS84_CRS
+from rio_tiler.errors import InvalidAssetName, MissingAssets
+from rio_tiler.io import BaseReader, MultiBaseReader, Reader
+from rio_tiler.io.stac import DEFAULT_VALID_TYPE, _get_assets
+from rio_tiler.types import AssetInfo
+
+
+@attr.s
+class PgSTACReaderAlt(MultiBaseReader):
```

Comment:
Q: why do we need to copy most of the code from titiler-pgstac? 

can't we just do 
```python
from titiler.pgstac.reader import PgSTACReader

@attr.s
class PgSTACReaderAlt(PgSTACReader):

    def _get_asset_info(self, asset: str) -> AssetInfo:
        """Validate asset names and return asset's url.
        Args:
            asset (str): STAC asset name.
        Returns:
            str: STAC asset href.
        """
        if asset not in self.assets:
            raise InvalidAssetName(f"{asset} is not valid")

        asset_info = self.input.assets[asset]
        extras = asset_info.extra_fields

        if ("alternate" not in extras) or ("s3" not in extras["alternate"]):
            raise MissingAssets("No alternate asset found")

        info = AssetInfo(url=extras["alternate"]["s3"]["href"], metadata=extras)

        if "file:header_size" in asset_info.extra_fields:
            h = asset_info.extra_fields["file:header_size"]
            info["env"] = {"GDAL_INGESTED_BYTES_AT_OPEN": h}

        if requester_pays := extras["alternate"]["s3"].get("storage:requester_pays"):
            if requester_pays:
                info["env"] = {"AWS_REQUEST_PAYER": "requester"}

        if bands := extras.get("raster:bands"):
            stats = [
                (b["statistics"]["minimum"], b["statistics"]["maximum"])
                for b in bands
                if {"minimum", "maximum"}.issubset(b.get("statistics", {}))
            ]
            if len(stats) == len(bands):
                info["dataset_statistics"] = stats

        return info
```

Reply:
My memory on this was fuzzy but I thought I recalled strange inheritance behavior with attr properties. I'll clean up the duplication